lib/pbench/test/unit/server/test_visualize.py

```diff
@@ -104,13 +105,16 @@ def mock_extract_data(self, test_name, dataset_name, input_type, data):
         )
         monkeypatch.setattr(QuisbyProcessing, "extract_data", mock_extract_data)
         response = query_get_as("uperf_1", "test", HTTPStatus.INTERNAL_SERVER_ERROR)
-        response.json["message"] = "Unexpected failure from Quisby"
+        assert response.json.get("message").startswith(
+            "Internal Pbench Server Error: log reference "
+        )
 
     def test_unsupported_benchmark(self, query_get_as, monkeypatch):
         def mock_get_metadata(self, dataset, key):
-            return {"dataset.metalog.pbench.script": "linpack"}
+            return {"dataset.metalog.pbench.script": "hammerDB"}
 
         monkeypatch.setattr(CacheManager, "find_dataset", self.mock_find_dataset)
         monkeypatch.setattr(ApiBase, "_get_dataset_metadata", mock_get_metadata)
+        monkeypatch.setattr(QuisbyProcessing, "extract_data", self.mock_extract_data)
```

Comment:
The `extract_data()` mock which you've added here should never be called by this test, right?  So, there's no point in burdening this test with the setup for it.  (I think you misunderstood what I was suggesting in [my earlier comment](https://github.com/distributed-system-analysis/pbench/pull/3463#discussion_r1234449489).)

What _would_ be useful is to have a mock for `extract_data()` which sets a flag if it is called; then this test could assert that the flag never gets set, which would prove that the CUT is properly avoiding the call to Quisby when the `results.csv` file is missing.